sklearn/utils/estimator_checks.py

```diff
@@ -947,10 +957,15 @@ def check_array_api_input(
         "transform",
     )
 
+    methods_to_skip = skips.get(array_namespace, [])
+
     for method_name in methods:
         method = getattr(est, method_name, None)
-        if method is None:
-            continue
+        if method is None or method_name in methods_to_skip:
+            raise SkipTest(
+                f"{array_namespace} is not array-API compliant for method"
```

Comment:
I think we tend to use the title case convention when mentioning the Array API standard.

```suggestion
                f"{array_namespace} is not Array API compliant for method"
```